packages/devtools_app/lib/src/screens/logging/logging_screen_v2/logging_table_row.dart

```diff
@@ -112,3 +157,147 @@ class _LoggingTableRowState extends State<LoggingTableRow> {
     );
   }
 }
+
+@visibleForTesting
+abstract class MetadataChit extends StatelessWidget {
```

Comment:
instead of using an abstract base class, can just use composition instead? Or use a non-abstract base class that we can setup properly with parameters. That will result in a lot less boiler plate to just have a class like this that we can build off of:
```
class MetadataChip extends StatelessWidget {
    const MetadataChit({
    super.key,
    required this.value,
    required this.icon,
    ...
    required this.maxWidth,
  });

  final String value;
  final IconData icon;
  ...
  final double maxWidth;
}
```

Reply:
I'm not sure I understand what is being asked.  Could you please expand?

The reason why I picked this route is it is similar to the column approach that was done previously, but still allows us to keep any complex calculations inside the extending class. Since there will be more types of chips being added with more complex stuff I figured it would be better to keep that stuff wrapped up nicely so we can just pass the LogDataV2 instead picking it apart ourselves.

Reply:
Take a look at the GaDevToolsButton class as an example. We extend this class with several other classes but we don't override a bunch of methods. We pass things through as parameters instead. This is more idiomatic flutter. The ColumnData class is a bit different because that class is not a Widget, but rather a Dart data class.

Reply:
I started trying to do this today but ran into some issues. Since these are stateful widgets 
If I try to pass the value arguments in the constructor, then I get crap from the compiler because the value isn't const (See images below). To overcome this I think I would have to do some complicated inheritance of the widget and the state.

Would you be ok with me leaving the current implementation and having someone who knows more about this do the refactor if it is important?

![image](https://github.com/user-attachments/assets/ab4b8ea3-352e-4aad-a18e-d39854a290c1)
![image](https://github.com/user-attachments/assets/8492a38e-9100-43f0-b806-192163936316)

Reply:
I pushed a commit to the PR. Just needed to remove `const` and restructure where the elapsed time was calculated for the `FrameElapsedMetaDataChip`.